notebooks/ch12/LR_Case_Study.md

```diff
@@ -0,0 +1,792 @@
+
+
+
+```python
+X = (drinking_data
+     .loc[:, ['Bias', 'Urban population', 'Late births', 'Wine', 'Liquor' ]]
+     .as_matrix())
+```
+
+`y` consists of the `Cirrhosis` column of our dataset.
+
+
+```python
+y = drinking_data['Cirrhosis'].as_matrix()
+```
+
+Now we just need to call the `minimize` function defined in a previous section.
+
+
+```python
+thetas = minimize(mse_cost, grad_mse_cost, X, y)
+```
+
+    theta: [ 0.  0.  0.  0.  0.] | cost: 4217.20
+    theta: [ 0.01  0.64  0.45  0.14  0.63] | cost: 999.01
+    theta: [ 0.    1.    0.39  0.59  0.21] | cost: 955.60
+    theta: [-0.02  0.86  0.1   1.58  0.34] | cost: 902.97
+    theta: [-0.01  0.41  0.82  2.13  0.13] | cost: 809.45
+    theta: [-0.08  0.55  0.54  1.86  0.19] | cost: 634.59
+    theta: [-0.33  0.43  0.63  1.86  0.11] | cost: 339.56
+    theta: [-1.26  0.22  0.76  1.8  -0.01] | cost: 98.33
+    theta: [-3.18  0.2   0.85  1.81 -0.02] | cost: 97.67
+    theta: [-7.49  0.17  1.01  1.81 -0.04] | cost: 96.69
+    theta: [-11.19   0.14   1.16   1.81  -0.05] | cost: 96.43
+    theta: [-11.19   0.14   1.16   1.81  -0.05] | cost: 96.43
+    theta: [-11.19   0.14   1.16   1.81  -0.05] | cost: 96.43
+    theta: [-11.19   0.14   1.16   1.81  -0.05] | cost: 96.43
+    
+
+Looks like gradient descent converged to those theta values! Thus, our linear model is:
+
+$y = -11.19 + 0.14x_1 + 1.16x_2 + 1.81x_3 - 0.05x_4$
+
+Let's compare this equation that we obtained to the one we would get if we had used `sklearn`'s LinearRegression model instead.
+
+
+```python
+model = LinearRegression()
+model.fit(X, y)
+model.coef_
+```
+
+
+
+
+    array([ 0.  ,  0.14,  1.16,  1.81, -0.05])
+
+
+
+The coefficients look pretty similar! Our homemade functions create the same model as an established Python package!
+
+## Evaluating our Model
+
+Now that we have a model, let's see how well it does on our dataset by plotting a residual plot of the percent error between the predicted Cirrhosis death rate values and the actual Cirrhosis death rate values.
+
+
+```python
+# HIDDEN
+resid = y - linear_model(thetas, X)
+resid_prop = resid / drinking_data['Cirrhosis']
+plt.scatter(np.arange(len(resid_prop)), resid_prop, s=15)
+plt.axhline(0)
+plt.title('Residual proportions (resid / actual Cirrhosis death rate)')
+plt.xlabel('Index of row in data')
+plt.ylabel('Error proportion');
+```
+
+
+![png](LR_Case_Study_files/LR_Case_Study_51_0.png)
+
+
+Looks like our linear model does pretty well when it predicts something higher than the actual value (it is usually within 25% of the correct value), but our model can do pretty abysmally when it predicts something lower than the actual value (there are a good number of error proportions around and above 30%). 
+
+As we can tell from the residual plot, our model is not excellent. We can hypothesize why our model is not doing well:
```

Comment:
I read excellent at first; I think its generally better to stay away from negative-positive and just use negative or positive.

we may want to combine this with the previous paragraph so we have a tighter analysis, since they say the same thing but referring back to the residual plot after already interpreting it is confusing